src/platforms/java/common/performance/index.mdx

```diff
@@ -166,3 +166,50 @@ if (span != null) {
     }
 }
 ```
+
+### Connecting Errors with Spans
+
+Sentry errors can be linked with transactions and spans.
+
+Errors reported to Sentry while transaction or span is running are linked automatically:
+
+```java
+ISpan span = Sentry.startTransaction("task");
+try {
+    processItem();
+} catch (Exception e) {
+    Sentry.captureException(e);
+} finally {
+    span.finish();
+}
+```
+
+In cases where the decision if thrown exception should be reported to Sentry happens later in the future after span is already finished, to connect error with a span you must:
+
+1. Link exception with a Span by calling `setThrowable` method:
+
+```java
+ISpan span = Sentry.getSpan();
+try {
+    processItem();
+} catch (Exception e) {
+    span.setThrowable(e);
+    throw e;
+} finally {
+    span.finish();
+}
+
+```
+
+2. Retrieve span context associated with exception when reporting an error with `IHub#getSpan`:
+
+```java
+SentryEvent event = new SentryEvent(throwable);
+SpanContext spanContext = HubAdapter.getInstance().getSpan(ex);
```

Comment:
If we're going for `HubAdapter` it means we need to rethink our API here.

Is `getSpan` supposed to be on the main `Sentry` class? 

Better:

Can we delete this section and simply have that in the docs? When we capture an exception we lookup for a span in the weakref? don't think this should be a user concern

Reply:
This means that we would need to move `WeakHashMap<Throwable, SpanContext> throwableToSpanContext` from `Hub` to `SentryClient` so that it can be applied for `captureEvent`, `captureEvent` and `captureException`. Which is fine for me but I am not sure if it does not violate any rules regarding what can be inside `SentryClient` cc @marandaneto

Reply:
`HubAdapter` is only meant for easing unit tests, not to be used externally, so definitely we'd need to rethink the API.

https://develop.sentry.dev/sdk/unified-api/
Hub should manage the state and `throwableToSpanContext` looks like managing the state.
Client is responsible for building the events only, no state at all afaik.

I didn't have the chance to read/test the whole transaction feature, not sure if I'm already able to give a better option, but working on it.

Reply:
> Sorry but I don't follow:
> Why does it need to go to the client? The API entry point is `Sentry.captureX` which goes through the Hub. We don't want to have `getSpan` there? Is this method not in the spec?
> 
> Lets just remove this from the docs for now, this was created to support our Spring integration and doesn't need to be documented yet.

yeah I also don't follow, don't really understand why we need that, I'm blind here.